website/docs/constructs/cookbook/construct-design.md

```diff
@@ -0,0 +1,86 @@
+---
+sidebar_position: 1
+sidebar_label: Constructs design
+---
+
+# Building new constructs
+
+AWS DSF Constructs follow the best practices detailed below. We recommend you follow them for building any new construct.
+
+## Construct design
+
+Refer to these best practices when design a new construct:
+
+* Use AWS CDK L2 Constructs instead of L1 when available. Alpha constructs should not be used because there may be some breaking changes in the API.
+
+* Resources naming should be optional and CDK should generate them when not provided. CDK ensures names uniqueness. If the name of the resource is provided as part of the constructor props, the name should be unique at least in the stack, sometimes in the AWS account or globally (like S3 bucket names) depending on the type of the resource.
+
+* Delete retention policy should be `RETAIN` by default but the possibility to `DESTROY` resources when destroying the CDK stack should be provided in the construct Props. This DESTROY parameter should only be considered (resources to be deleted) if the global guardrail is configured accordingly (global CDK parameter to be defined). The overall guard rail approach is explained in the [`DataLakeStorage`](../library/02-Storage/03-data-lake-storage.mdx#objects-removal) and code example is available [here](https://github.com/awslabs/aws-data-solutions-framework/blob/main/framework/src/storage/lib/data-lake-storage.ts#L150).
+  
+* Extend `Construct` vs L2 Resource: 
+    * If the construct you want to create contains only one resource and it’s a specific implementation/configuration, then you should go with extending the base resource. An example is the [`AnalyticsBucket`](https://github.com/awslabs/aws-data-solutions-framework/blob/main/framework/src/storage/lib/analytics-bucket.ts#L33).
+    * If the resource contains more than one resource, it should extend `Construct`. It allows to properly scope resources created within the construct (using `this` parameter as the scope) and avoid collisions in resource naming.
+
+
+* Expose all the resources by your construct so you provide a way to end-users to [customize](../customize.mdx) them. Sometimes, it's more user friendly to expose a resource directly to avoid long resources chaining. For example, in the `DataLakeStorage` construct that contains a KMS `Key` attached to the `AnalyticsBucket` part of the `DataLakeStorage`, we re-expose the key so you can access it directly via `dataLakeStorage.key` instead of `dataLakeStorage.bronze.key`.
```

Comment:
```suggestion
* Expose all the resources created by your construct so you provide a way for end-users to [customize](../customize.mdx) them. Sometimes, it's more user friendly to expose a resource directly to avoid long resources chaining. For example, in the `DataLakeStorage` construct that contains a KMS `Key` attached to the `AnalyticsBucket` part of the `DataLakeStorage`, we re-expose the key so you can access it directly via `dataLakeStorage.key` instead of `dataLakeStorage.bronze.key`.
```